### PR TITLE
Switch to using TargetFrontendConfig in the frontend memory tests

### PR DESCRIPTION
**Pull request by abrown**
~~Commit 7d36fd9a1e070e2d83f8b63e44faae587a138686 avoided these
x86-specific tests altogether. This change attempts to run the tests on
whichever backend is native to the current host system.~~

[edit: based on @bjorn3's suggestion, here is a slightly different approach. I think this resolves the concerns about not running these tests on certain systems. Also, it factors out some duplicated code.]

~~Commit 7d36fd9 avoided these
x86-specific tests altogether. This change forces the x86 backend to be
present when the frontend tests are run.~~

[edit: @bjorn3's final suggestion is actually the best--just avoid any backend-specific dependency entirely]

Commit 7d36fd9 avoided these
x86-specific tests altogether. This change avoids any dependency on x86
entirely by specifying a frontend configuration (SystemV + U64); this is
enough information for the `FunctionBuilder` to correctly generate the
syscalls.

### CODE REVIEW

**Review by cfallin — APPROVED** [review]
Thanks!

**Comment by bjorn3** [comment]
This is only correct for 64bit systems using the SystemV call conv as default.

**Comment by abrown** [comment]
> This is only correct for 64bit systems using the SystemV call conv as default.

Yup. How is the current `#[cfg_attr(not(target_arch = "x86_64"), ignore)]` correct, though? When run on a non-SystemV-OS, I would guess the test should run into similar calling convention errors...

**Comment by bjorn3** [comment]
Previously plain `x86_64` was used as target, which always defaults to SystemV. After this change the OS is also included, which can override the default calling convention. An alternative to this PR would be to add a dev-dependency on cranelift-codegen to cranelift-frontend with the x86_64 backend enabled.

File: cranelift/frontend/src/frontend.rs [hunk 1]
```diff
@@ -968,24 +969,21 @@ mod tests {
...
-            .ok()
-            .map(|b| b.finish(shared_flags))
-            .expect("This test requires x86_64 support.");
-
-        let mut sig = Signature::new(target.default_call_conv());
+    #[cfg_attr(not(target_pointer_width = "64"), ignore)]
+    #[test]
+    fn memcpy() {
+        let target = host_target();
+        let mut sig = Signature::new(CallConv::SystemV);
```
Comment (bnjbvr) [inline cranelift/frontend/src/frontend.rs:986]: I think you could get the default calling convention from the `TargetIsa` object, and then use it to generate the calling convention in the expected string below. Not sure if it works for all the cases, though.

**Review by bnjbvr — COMMENTED** [review]
Interesting approach!

**Comment by bjorn3** [comment]
A final suggestion would be to manually construct a specific `FrontendConfig` value and then avoid `TargetIsa` completely.

File: cranelift/frontend/src/frontend.rs (hunk 1 above)
Comment (abrown) [reply cranelift/frontend/src/frontend.rs:988]: Ok, I ended up changing the approach... I am just trying various things based on the discussion [here](https://bytecodealliance.zulipchat.com/#narrow/stream/217126-wasmtime/topic/testing).

**Review by bjorn3 — APPROVED** [review]

**Review by cfallin — APPROVED** [review]
This is much much nicer -- thanks!

**Pull request merged**